File: app/src/main/java/com/scatl/uestcbbs/module/home/adapter/OnLineUserAdapter.java
package com.scatl.uestcbbs.module.home.adapter;

import android.widget.ImageView;

import androidx.annotation.NonNull;

import com.bumptech.glide.Glide;
import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.scatl.uestcbbs.R;
import com.scatl.uestcbbs.entity.OnLineUserBean;

/**
 * author: sca_tl
 * date: 2020/10/7 11:32
 * description:
 */
public class OnLineUserAdapter extends BaseQuickAdapter<OnLineUserBean.UserBean, BaseViewHolder> {
    public OnLineUserAdapter(int layoutResId) {
        super(layoutResId);
    }

    @Override
    protected void convert(@NonNull BaseViewHolder helper, OnLineUserBean.UserBean item) {
        helper.setText(R.id.item_online_user_name, item.userName)
                .setText(R.id.item_online_user_time, item.time);
        Glide.with(mContext).load(item.userAvatar).into((ImageView) helper.getView(R.id.item_online_user_avatar));
    }
}

File: app/src/main/java/com/scatl/uestcbbs/module/home/adapter/OnLineUserAdapter.kt
package com.scatl.uestcbbs.module.home.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import com.scatl.uestcbbs.databinding.ItemOnlineUserBinding
import com.scatl.uestcbbs.entity.OnLineUserBean
import com.scatl.uestcbbs.helper.PreloadAdapter
import com.scatl.uestcbbs.helper.ViewBindingHolder
import com.scatl.uestcbbs.util.load

/**
 * Created by sca_tl at 2023/6/9 16:38
 */
class OnLineUserAdapter: PreloadAdapter<OnLineUserBean.UserBean, ItemOnlineUserBinding>() {

    override fun getViewBinding(parent: ViewGroup): ItemOnlineUserBinding {
        return ItemOnlineUserBinding.inflate(LayoutInflater.from(context), parent, false)
    }

    override fun onBindViewHolder(holder: ViewBindingHolder<ItemOnlineUserBinding>, position: Int, item: OnLineUserBean.UserBean?) {
        super.onBindViewHolder(holder, position, item)
        if (item == null) {
            return
        }

        holder.binding.userName.text = item.userName
        holder.binding.activeTime.text = item.time
        holder.binding.avatar.load(item.userAvatar)
    }
}